common/src/main/java/org/opensearch/ml/common/connector/Connector.java

```diff
@@ -77,21 +81,37 @@ public interface Connector extends ToXContentObject, Writeable {
 
     <T> void parseResponse(T orElse, List<ModelTensor> modelTensors, boolean b) throws IOException;
 
-    default void validatePayload(String payload) {
+    default Set<String> getRequiredParameters(String payload) {
+        Set<String> requiredParameters = new HashSet<>();
         if (payload != null && payload.contains("${parameters")) {
             Pattern pattern = Pattern.compile("\\$\\{parameters\\.([^}]+)}");
             Matcher matcher = pattern.matcher(payload);
-
-            StringBuilder errorBuilder = new StringBuilder();
             while (matcher.find()) {
                 String parameter = matcher.group(1);
-                errorBuilder.append(parameter).append(", ");
+                requiredParameters.add(parameter);
             }
-            String error = errorBuilder.substring(0, errorBuilder.length() - 2).toString();
+        }
+        return requiredParameters;
+    }
+
+    default void validateParameters(Set<String> requiredParameters, Map<String, String> parameters) {
+        StringBuilder errorBuilder = new StringBuilder();
+        for (String requiredParameter : requiredParameters) {
+            if (!parameters.containsKey(requiredParameter)) {
+                errorBuilder.append(requiredParameter).append(", ");
+            }
+        }
+        if (!errorBuilder.isEmpty()) {
+            String error = errorBuilder.substring(0, errorBuilder.length() - 2);
             throw new IllegalArgumentException("Some parameter placeholder not filled in payload: " + error);
         }
     }
 
+    default void validatePayload(String payload) {
```

Comment:
This default method doesn't make sense, please change this to an abstract method.

Reply:
Sorry for not getting your point. This method is used when parameter RECURSIVE_PARAMETER_ENABLED is true, which is the default behavior.

Reply:
When adding a new default function, basically you assume that this function will be used in multiple sub-classes, and only a few overrides it, this doesn't seem the case since only one place using it.
Also, there's already the `validateParameters` method which is basically same function with this method, it's redundant to add a similar function in interface layer.

Reply:
Ack. Will update this.